libs/libcommon/include/common/DateLUT.h

```diff
@@ -10,9 +10,7 @@
 #include <mutex>
 #include <memory>
 
-/** Этот класс предоставляет метод для того, чтобы создать объект DateLUTImpl
-  * для заданного часового пояса, если он не существует.
-  */
+/// This class provides lazy initialization and lookup of singleton DateLUTImpl objects for a given timezone.
 class DateLUT : public Singleton<DateLUT>
```

Comment:
I just noticed, that existing implementation of METR-15618 is both:
- not optimal;
- not correct.

```
auto ti = convertTimestamp(vec_from[i], remote_date_lut, local_date_lut);
FormatImpl<DataTypeDateTime>::execute(ti, write_buffer);
```
- totally incorrect :(

```
:) SELECT 1301146200 + 1800 * number AS ts, toString(toDateTime(ts), 'Australia/Sydney') AS time_in_sydney FROM system.numbers LIMIT 7

SELECT 
    1301146200 + (1800 * number) AS ts, 
    toString(toDateTime(ts), 'Australia/Sydney') AS time_in_sydney
FROM system.numbers 
LIMIT 7

┌─────────ts─┬─time_in_sydney──────┐
│ 1301146200 │ 2011-03-27 00:30:00 │
│ 1301148000 │ 2011-03-27 01:00:00 │
│ 1301149800 │ 2011-03-27 01:30:00 │
│ 1301151600 │ 2011-03-27 01:00:00 │
│ 1301153400 │ 2011-03-27 01:30:00 │
│ 1301155200 │ 2011-03-27 03:00:00 │
│ 1301157000 │ 2011-03-27 03:30:00 │
└────────────┴─────────────────────┘
```

Because it is done through conversion to moscow time and then formatting in moscow time. The correct (and also more optimal) way is to directly format using specified DateLUT.

That's why I don't care about 8% right now. We should fix that in separate task.